src/pages/core/architecture/actor-model.mdx

```diff
@@ -0,0 +1,58 @@
+---
+tags: ["architecture", "core"]
+---
+
+import Image from "next/image";
+import { Callout } from "nextra/components";
+
+# Actor Model
+
+CosmWasm, at its core, is built around the actor model. This prevents some
+common pitfalls which, for example, Ethereum smart contracts had to fix over the
+years. The issues mostly revolved around
+[reentrancy issues](https://ethereum.org/en/developers/docs/smart-contracts/security/#reentrancy)
+where we could, for example, call out to another contract before finishing the
+current execution.
+
+<Callout>
+  If you are already familiar with the actor model through, for example,
+  programming in languages such as Erlang or Elixir, you can skip this page.
+</Callout>
+
+While the term "actor model" sounds fancy, it is actually quite simple. Each
+contract is an "actor" and _plays_ a role in the system (pun intended).
+
+Each actor has its own state and can only be interacted with via messages. This
+means a contract can only interact with the outside world via messages and only
+manipulate its own state.
+
+Picture two people living in two houses, each house has a mailbox. If person A
```

Comment:
Been a while since I've used snail mail, but, I believe I would put the letter in my own mailbox, then the postal service picks it up to put it in the other person's.

Kinda a nit, since the point is more about addresses and messages than delivery - but  when adding messages in contracts, we don't actually add the message directly to the other contract either - we add it to _our_ Response object, and then it's picked up and delivered to the destination, so I think it would fit better overall to describe it this way.

Reply:
Dang, okay, I hit the merge button a little too quickly. Yeah, good point, I'll have a look over the wording again and make a separate PR for that.